components/layout_2020/geom.rs

```diff
@@ -526,16 +540,12 @@ impl From<LogicalRect<CSSPixelLength>> for LogicalRect<Au> {
     }
 }
 
-/// Convert a `PhysicalRect<Length>` (one that uses CSSPixel as the unit) to an untyped `Rect<Au>`.
-pub fn physical_rect_to_au_rect(rect: PhysicalRect<Length>) -> euclid::default::Rect<Au> {
+pub fn au_rect_to_length_rect(rect: &Rect<Au, UnknownUnit>) -> euclid::default::Rect<Length> {
     euclid::default::Rect::new(
-        euclid::default::Point2D::new(
-            Au::from_f32_px(rect.origin.x.px()),
-            Au::from_f32_px(rect.origin.y.px()),
-        ),
+        euclid::default::Point2D::new(Length::from(rect.origin.x), Length::from(rect.origin.y)),
         euclid::default::Size2D::new(
-            Au::from_f32_px(rect.size.width.px()),
-            Au::from_f32_px(rect.size.height.px()),
+            Length::from(rect.size.width),
+            Length::from(rect.size.height),
         ),
     )
 }
```

Comment:
I would move `stacking_context.rs` since it's only used there and it doesn't deal at all with Layout 2020's geometry types. Additionally, I would also import the `euclid::default::{Rect, Point2D, Size2D}` as they are just a named type of `Rect<T, UnknownUnit>`. Then this could just be:

```rust
pub fn au_rect_to_length_rect(rect: &Rect<Au>) -> Rect<Length> {
    Rect::new(
        Point2D::new(rect.origin.x.into(), rect.origin.y.into()),
        Size2D::new(rect.size.width.into(), rect.size.height.into()),
    )
}
```